packages/vite/src/node/optimizer/scan.ts

```diff
@@ -315,21 +315,32 @@ function esbuildScanPlugin(
         let ext = path.extname(id).slice(1)
         if (ext === 'mjs') ext = 'js'
 
+        let loader = ext as Loader
+
+        // Use the user given loader if it is explicitly set
+        if (config.esbuild && config.esbuild.loader) {
```

Comment:
**suggestion (if-minor):** I think that can be shortened to 
```suggestion
        if (config.esbuild?.loader) {
```

Reply:
`config.esbuild` is of type `false | ESBuildOptions | undefined`, but the optional chaining `?.` will not narrow the types because of the `false` here.

> Property 'loader' does not exist on type 'false | ESBuildOptions'.

Reply:
Yeah sorry for such inconvenient reviews, but I review in the GitHub Web UI and there I don't have type-checking
I hope we can improve this with #2613, then eslint will show up warnings in the Web UI via #2622